### DISCUSSION TITLE
Using `extend type` with GraphQL API

### DISCUSSION THREAD

**Discussion by bombillazo** [graphql/graphql-js#3266] [Q&A] [answered]
Hello, how can I represent the following SDL schema :
```gql
type MyType {
  prop: Int
}

extend type MyType {
  newProp: String
}
```
With the `GraphQLObjectType` API?

I have different files where I programmatically create my schema and want a "common" Query and Mutation type from which all modules extend to add their functions.

**Comment by saihaj** [comment]
Don't think it is possible yet https://github.com/graphql/graphql-js/issues/1569

**Comment by IvanGoncharov** [comment, marked as answer]
With
```js
const myType = new GraphQLObjectType({
  name: 'MyType',
  fields: {
    'prop': { type: GraphQLInt },
  },
});
```
you can do the following:
```js
const myTypeConfig = myType.toConfig();

const exptendedMyType = new GraphQLObject({
  ...myTypeConfig,
  fields: {
    ...myTypeConfig.fields,
    newProp: { type: GraphQLString },
  },
});
```

**Reply by bombillazo** [reply]
Wow, very nice!
